source/globals.cpp

```diff
@@ -99,59 +100,103 @@ void parse_settings_cfg(){
           }
 
-          if (line.rfind("username", 0) == 0) {
-            srv.rucio_conn_params.user_name = get_cfg_value(line);
+          auto ca_file = fopen(ca_file_path.data(), "rb");
+          if (!ca_file) {
+            fastlog(ERROR, "CA file not found at %s. Server will be skipped.", ca_file_path.data());
+            continue;
           }
 
-          if (line.rfind("password", 0) == 0) {
-            srv.rucio_conn_params.password = get_cfg_value(line);
-          }
-
-          if (line.rfind("account", 0) == 0) {
-            srv.rucio_conn_params.account_name = get_cfg_value(line);
+          fastlog(INFO, "\tServer %d -> %s:\n"
+                        "\t\turl = %s\n"
+                        "\t\taccount = %s\n"
+                        "\t\tusername = %s\n"
+                        "\t\tpassword = %s",
+                  i_srv++,
+                  srv_name.data(),
+                  srv.rucio_conn_params.server_url.data(),
+                  srv.rucio_conn_params.account_name.data(),
+                  srv.rucio_conn_params.user_name.data(),
+                  srv.rucio_conn_params.password.data());
+
+          rucio_server_map.emplace(std::make_pair(srv_name, srv));
+
+          if (not rucio_validate_server(srv_name)) {
+            fastlog(ERROR, "Unable to validate server %s. Dropping.", srv_name.data());
+          } else {
+            rucio_server_names.emplace_back(srv_name);
+            fastlog(INFO, "Server %s validated!", srv_name.data());
           }
         }
-
-        fastlog(INFO, "\tServer %d -> %s:\n"
-                      "\t\turl = %s\n"
-                      "\t\taccount = %s\n"
-                      "\t\tusername = %s\n"
-                      "\t\tpassword = %s",
-                i_srv++,
-                srv_name.data(),
-                srv.rucio_conn_params.server_url.data(),
-                srv.rucio_conn_params.account_name.data(),
-                srv.rucio_conn_params.user_name.data(),
-                srv.rucio_conn_params.password.data());
-
-        rucio_server_map.emplace(std::make_pair(srv_name,srv));
-
-        if(not rucio_validate_server(srv_name)){
-          fastlog(ERROR, "Unable to validate server %s. Dropping.", srv_name.data());
-        } else {
-          rucio_server_names.emplace_back(srv_name);
-          fastlog(INFO, "Server %s validated!", srv_name.data());
-        }
       }
-      inode = readdir(dp);
     }
   }
 }
+
+bool check_permissions(const std::string& mountpoint_path){
+  int mountpoint_status = access(mountpoint_path.data(), R_OK && W_OK);
+
+  if(mountpoint_status != 0){
+    DIR *dp = opendir(mountpoint_path.data());
+    if(dp) {
+      fastlog(ERROR, "Mountpoint at %s has wrong permissions!", mountpoint_path.data());
+      closedir(dp);
+    } else {
+      fastlog(ERROR, "Mountpoint at %s does not exist!", mountpoint_path.data());
+    }
+
+    return false;
+  }
+
+  int cache_status = access(rucio_cache_path.data(), R_OK && W_OK);
```

Comment:
This section is really neat! And was much-needed logging!
Thanks for adding this :smile: